app/models/chat_message.rb

```diff
@@ -49,6 +44,45 @@ def push_notification_excerpt
     message[0...400]
   end
 
+  def add_flag(user)
+    reviewable = ReviewableChatMessage.needs_review!(
+      created_by: user,
+      target: self,
+    )
+    reviewable.update(target_created_by: self.user)
+    reviewable.add_score(
+      user,
+      ReviewableScore.types[:needs_approval],
```

Comment:
I'm not sure "Needs Approval" is the correct type here. Approval refers to a specific thing in our app - when a post or other thing needs to be approved before it is posted. 

I would suggest adding a new score type. Maybe `review_requested` or `needs_review` or similar?